### PR TITLE
Add posts to partner details page & abstract post list

### LINKED ISSUE #947: Posts | Add Posts to the Partner Details page
This will follow the same implementation as Posts on Projects do. The only difference is the only "shareabilities" are:
- External
- Ask to Share Externally
- Internal

### PR DESCRIPTION
**Pull request by daniel-leod**
Closes #947

### CODE REVIEW

File: src/components/posts/PostForm/PostForm.tsx [hunk 1]
```diff
@@ -21,10 +22,19 @@ import { PostFormFragment } from './PostForm.generated';
 export type PostFormProps<T, R = void> = DialogFormProps<T, R> & {
   /** The pre-existing post to edit */
   post?: PostFormFragment;
+  disableMembership: boolean;
```
Comment (CarsonF) [inline src/components/posts/PostForm/PostForm.tsx:25]: Let's default this to false. Or if false is the edge case, let's change it to `includeMembership` - still defaulted to false.

File: src/components/posts/PostForm/PostForm.tsx [hunk 2]
```diff
@@ -21,10 +22,19 @@ import { PostFormFragment } from './PostForm.generated';
...
 };
 
+const shareabilityList = (disableMembership: boolean) =>
+  without(
+    PostShareabilityList,
+    'ProjectTeam',
```
Comment (CarsonF) [inline src/components/posts/PostForm/PostForm.tsx:31]: `PostShareabilityList` shouldn't include `ProjectTeam` anymore since it's been deprecated.

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/scenes/Partners/Detail/PartnerPostList.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      <List
+        {...posts}
+        classes={{
+          container: classes.postListItems,
+        }}
+        spacing={3}
+        renderItem={(post) => (
+          <PostListItemCard
+            disableMembership={true}
+            parent={partner as any}
```
Comment (CarsonF) [inline src/scenes/Partners/Detail/PartnerPostList.tsx:60]: What's the error you are getting here?

File: src/components/posts/PostList.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+const useStyles = makeStyles(() => ({
+  postListItems: {
+    maxWidth: 600,
+  },
+}));
+
+interface PostListProps {
+  parent: PostableIdFragment;
+  posts: any;
```
Comment (daniel-leod) [inline src/components/posts/PostList.tsx:21]: @CarsonF What would be the correct type here?

Comment (CarsonF) [reply src/components/posts/PostList.tsx]: I resolved with the latest commit.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**